sympy/integrals/tests/test_integrals.py

```diff
@@ -2098,6 +2098,17 @@ def test_issue_20782():
     assert integrate(f, (x, -oo, 1)) == 1
     assert integrate(-f, (x, -oo, 1)) == -1
 
+def test_issue_20781():
+    x_d = symbols('x_d')
+    x = symbols('x')
+    fun_sum = lambda x,a1,a2: Piecewise((0,x<a1),(1,x>=a1)) + Piecewise((0,x<a2),(1,x>=a2))
+
+    assert integrate(fun_sum((x_d),0,0.0),(x_d,-float('Inf'),x)) == 2*x - 2*Min(0.0, x)
+    assert integrate(fun_sum((x_d),0,0.0),(x_d,-float('Inf'),x)) == 2*x - 2*Min(0.0, x)
+    assert integrate(fun_sum((x_d),0.0,0),(x_d,-float('Inf'),x)) == 2*x - 2*Min(0.0, x)
+    assert integrate(fun_sum((x_d),0.0,0),(x_d,-float('Inf'),x)) == 2*x - 2*Min(0.0, x)
+    assert integrate(fun_sum((x_d),1.0,1),(x_d,-float('Inf'),x)) == 2*x - 2*Min(0.0, x)
+    assert integrate(fun_sum((x_d),1.0,1),(x_d,-float('Inf'),x)) == 2*x - 2*Min(0.0, x)
```

Comment:
It appears that the output of the assert statements is not as expected.

Reply:
resolved, apologies for the typo